lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -289,6 +289,48 @@ class DecomposeAtenExpandOp : public OpRewritePattern<AtenExpandOp> {
 };
 } // namespace
 
+// Decompose torch.mean into: sum(x)/div(numTensorElements).
+namespace {
+class DecomposeAtenMeanOp : public OpRewritePattern<AtenMeanOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenMeanOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value input = op.self();
+    Value output = op.result();
+
+    Value dtype = rewriter.create<ConstantNoneOp>(loc);
+    Value sum = rewriter.create<AtenSumOp>(loc, output.getType(), input, dtype);
+    if (!sum)
+      return failure();
+
+    BaseTensorType inputTensorType = input.getType().cast<BaseTensorType>();
+    BaseTensorType outputTensorType = output.getType().cast<BaseTensorType>();
+    SmallVector<int64_t> sizes;
+    int64_t tensorSize = 1;
+    if (inputTensorType.hasSizes()) {
+      ArrayRef<int64_t> inputShape = inputTensorType.getSizes();
+      for (int64_t i : inputShape) {
+        if (i < 1)
+          return rewriter.notifyMatchFailure(op, "Invalid tensor size.");
+        tensorSize *= i;
+      }
+    } else
+      return rewriter.notifyMatchFailure(
+          op, "Only support input with known sizes.");
+
+    Value constTensorSize = rewriter.create<ConstantFloatOp>(
+        loc, rewriter.getF32FloatAttr((float)tensorSize));
+    Value divisorTensor = rewriter.create<PrimNumToTensorScalarOp>(
+        loc, outputTensorType, constTensorSize);
+    rewriter.replaceOpWithNewOp<AtenDivTensorOp>(op, output.getType(), sum,
```

Comment:
`AtenDivTensorOp ` doesn't support type promotion. We need to add the support here https://github.com/llvm/torch-mlir/blob/1dc374014b4539651fdf36c83b896b93fd5703a5/lib/Conversion/TorchToLinalg/TorchToLinalg.cpp#L1470-L1476. You could refer to https://github.com/llvm/torch-mlir/blob/1dc374014b4539651fdf36c83b896b93fd5703a5/lib/Conversion/TorchToLinalg/TorchToLinalg.cpp#L1428-L1442 about how to add type promotion support.